File: 3838-weighted-word-mapping/3838-weighted-word-mapping.py
class Solution:
    def mapWordWeights(self, words: List[str], weights: List[int]) -> str:
        res = ""
        for word in words:
            score = 0
            for ch in word:
                score += weights[ord(ch) - ord('a')]
            res += chr(ord('z') - (score % 26))
        return res

File: 3838-weighted-word-mapping/3838-weighted-word-mapping.cpp
class Solution {
public:
    string mapWordWeights(vector<string>& words, vector<int>& weights) {
        string res = "";
        for(int i = 0; i < words.size(); i++)
        {
            int score = 0;
            for(int j = 0; j < words[i].length(); j++)
            {
                score += weights[words[i][j] - 'a'];
            }
            res += ('z' - (score % 26));
        }
        return res;
    }
};